sound/soc/intel/boards/Kconfig

```diff
@@ -377,4 +377,18 @@ config SND_SOC_INTEL_SKL_HDA_DSP_GENERIC_MACH
 
 endif ## SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC || SND_SOC_SOF_HDA_AUDIO_CODEC
 
+if SND_SOC_SOF_CANNONLAKE || SND_SOC_SOF_ICELAKE
```

Comment:
this could also work on APL and GLK, so we'd probably need a Kconfig shortcut to say SKL+

Reply:
we can use SND_SOC_SOF_HDA_COMMOM for SKL+, or SND_SOC_SOF_INTEL_COMMON for all Intel platforms including byt.

Reply:
Yes indeed.
I am not sure we can reuse this for Baytrail, you have HDMI which is not present in Baytrail. And if you make it conditional on HDAC_HDMI, you could have a build with SKL+ included so would need an additional runtime detection. Let's stick with SKL+ for now.

Reply:
@plbossart Baytrail is my next plan.  I will add #if condition to make sure the machine driver can work without HDMI supported